models/entities/activity.go

```diff
@@ -0,0 +1,97 @@
+package entities
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/meowpub/meow/config"
+	"github.com/meowpub/meow/jsonld"
+	"github.com/meowpub/meow/lib"
+	"github.com/meowpub/meow/server/api"
+	"github.com/pkg/errors"
+)
+
+type Activity struct {
+	Object
+
+	Actor      jsonld.Ref `json:"https://www.w3.org/ns/activitystreams#actor,omitempty"`
+	Obj        jsonld.Ref `json:"https://www.w3.org/ns/activitystreams#object,omitempty"`
+	Target     jsonld.Ref `json:"https://www.w3.org/ns/activitystreams#target,omitempty"`
+	Origin     jsonld.Ref `json:"https://www.w3.org/ns/activitystreams#origin,omitempty"`
+	Result     jsonld.Ref `json:"https://www.w3.org/ns/activitystreams#result,omitemtpy"`
+	Instrument jsonld.Ref `json:"https://www.w3.org/ns/activitystreams#instrument,omitempty"`
+}
+
+var _ Entity = &Object{}
+
+var activityKind = &EntityKind{
+	Name: "activity",
+	Unmarshall: func(obj map[string]interface{}) (Entity, error) {
+		v := &Activity{}
+		return v, jsonld.Unmarshal(obj, v)
+	},
+	Marshall: func(e Entity) (map[string]interface{}, error) {
+		v, err := jsonld.Marshal(e.(*Activity))
+		if err != nil {
+			return nil, err
+		} else {
+			return v.(map[string]interface{}), nil
+		}
+	},
+}
+
+func (*Activity) GetKind() *EntityKind {
+	return activityKind
+}
+
+// api.Handler
+
+// Return ourselves
+func (o *Activity) HandleRequest(ctx context.Context, req *http.Request) api.Response {
+	d, err := o.Hydrate(ctx)
+	if err != nil {
+		return api.ErrorResponse(err)
+	}
+
+	return api.Response{
+		Data: d,
+	}
+}
+
+func (self *Activity) Hydrate(ctx context.Context) (map[string]interface{}, error) {
+	if o, err := jsonld.Marshal(self); err == nil {
+		return jsonld.Compact(lib.GetHttpClient(ctx), o.(map[string]interface{}), "", "https://www.w3.org/ns/activitystreams")
+	} else {
+		return nil, err
+	}
+}
+
+func NewActivity(store *Store, parent Entity, types []string) (*Activity, error) {
```

Comment:
What would `parent` be in this case? I'm guessing for a Create event for a Note, the parent would be the Person who made it?

Reply:
In the code attached here (which creates the person's Create activity), I was putting the activity as a child of the outbox itself

so you end up with `example.com/~person/outbox/83428934789`

you would probably use the person as the parent of the note.
```
Object: example.com/~person/34789234789
Activity: example.com/~person/outbox/837489234789
```